docs/crud-subject.rst

```diff
@@ -0,0 +1,32 @@
+************
+Crud Subject
+************
+.. _crud-subject:
+
+The Crud Subject is the class which is passed as the subject of all the events that the Crud plugin emits during it's
+execution. Depending on the action being scaffolded, and what it's working on the contents of the subject can be
+different.
+
+Core event subjects
+===================
+
+You can find many of the subject contents are included as part of the :ref:`Core Crud Events <core-crud-events>`
+documentation. This is because the subject of the event is very specific to the event being emitted.
+
+When dealing with listeners, you are able to manipulate the subject of the event in order to change Cruds behavior. Such
+as changing pagination, or adding extra conditions to a query.
+
+This is an example of the data passed in a ``beforeFind`` event subject.
+
+.. code-block:: phpinline
+
+    <?php
+    public function view($id)
+    {
+        $this->Crud->on('beforeFind', function (\Cake\Event\Event $event) {
+            $query = $event->subject()->query;
```

Comment:
I thought so, [but it doesn't exist](https://github.com/cakephp/cakephp/blob/master/src/Event/Event.php#L33-L38). So it would be accessed through a magic method. I don't like magic methods and I feel that an api which provides an accessor method it doing so for a reason. Hooking into or extending a magic method would be more complex than a verbose accessor method, but that's my 2¢